File: .skills/openclaw-skills/skills/churchtg7/power-search/scripts/brave-search.js
const fetch = require('node-fetch');

class BraveSearch {
  constructor(apiKey) {
    this.apiKey = apiKey;
    this.baseUrl = 'https://api.search.brave.com/res/v1/web/search';
  }

  async search(query, limit = 10) {
    try {
      const params = new URLSearchParams({
        q: query,
        count: Math.min(limit, 20),
      });

      const response = await fetch(`${this.baseUrl}?${params}`, {
        headers: {
          'Accept': 'application/json',
          'X-Subscription-Token': this.apiKey,
        },
      });

      if (!response.ok) {
        throw new Error(`Brave API error: ${response.status} ${response.statusText}`);
      }

      const data = await response.json();

      // Parse results from nested structure
      const results = (data.web?.results || []).slice(0, limit).map((result, index) => ({
        rank: index + 1,
        title: result.title,
        url: result.url,
        description: result.description,
        age: result.age,
      }));

      return {
        query,
        count: results.length,
        results,
      };
    } catch (error) {
      throw new Error(`Brave Search failed: ${error.message}`);
    }
  }
}

module.exports = BraveSearch;

File: .skills/openclaw-skills/skills/mryee2023/yee-brave-search/brave_search.py
#!/usr/bin/env python3
import os
import sys
import json
import urllib.request
import urllib.parse

# Brave Search API Configuration
# Reads from environment variable BRAVE_API_KEY
API_KEY = os.getenv("BRAVE_API_KEY")
BASE_URL = "https://api.search.brave.com/res/v1/web/search"

def search_brave(query, count=5):
    if not API_KEY:
        return {"error": "Environment variable BRAVE_API_KEY is not set."}
        
    params = urllib.parse.urlencode({"q": query, "count": count})
    url = f"{BASE_URL}?{params}"
    
    headers = {
        "Accept": "application/json",
        "X-Subscription-Token": API_KEY
    }
    
    try:
        req = urllib.request.Request(url, headers=headers)
        with urllib.request.urlopen(req) as response:
            if response.status == 200:
                data = json.loads(response.read().decode())
                results = []
                if "web" in data and "results" in data["web"]:
                    for item in data.get("web", {}).get("results", []):
                        results.append({
                            "title": item.get("title"),
                            "url": item.get("url"),
                            "description": item.get("description")
                        })
                return results
            else:
                return {"error": f"HTTP {response.status}"}
    except Exception as e:
        return {"error": str(e)}

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print(json.dumps({"error": "No query provided"}, indent=2))
        sys.exit(1)
        
    query = " ".join(sys.argv[1:])
    results = search_brave(query)
    print(json.dumps(results, indent=2, ensure_ascii=False))
